cheatsheets/Credential_Stuffing_Prevention_Cheat_Sheet.md

```diff
@@ -1,84 +1,114 @@
+- A new browser/device or IP address.
+- An unusual country or location.
+- Specific countries that are considered untrusted.
+- An IP address that appears on known blacklists.
+- An IP address that has tried to login to multiple accounts.
+- A login attempt that appears to be scripted rather than manual.
 
-## Defense Option 1: Multi-Factor Authentication
+Additionally, for enterprise applications, known trusted IP ranges could be added to a whitelist so that MFA is not required when users connect from these ranges.
 
-True multi-factor authentication is the best defense against Credential Stuffing attacks, but can have significant deployment and usability impacts, and so is frequently not a feasible option. If this defense is not feasible for your application, consider adopting as many of these other defenses as you can.
+# Alternative Defenses
 
-In order to balance security and usability, multi-factor authentication can be combined with other techniques to ask for 2nd factor only in special situations. For example, it can be combined with [device fingerprinting](cheatsheets/Credential_Stuffing_Prevention_Cheat_Sheet.md#defense-option-4-device-fingerprinting), in that scenario, the application can ask for a 2nd factor only when the application is accessed by an unknown, new browser. Similary, cookies can also be used for remembering known browsers or the application can ask about another factor based on the login success ratio, IP address (like users from different network than company one like remote workers) etc.
+Where it is not possible to implement MFA, there are many alternative defenses that can be used to protect against credential stuffing and password spraying. In isolation none of these are as effective as MFA, however if multiple defenses are implemented in a layered approach, they can provide a reasonable degree of protection. In many cases, these mechanisms will also protect against brute-force or password spraying attacks.
 
-## Defence Option 2: Use a CAPTCHA
+Where an application has multiple user roles, it may be appropriate to implement different defenses for different roles. For example, it may not be feasible to enforce MFA for all users, but it should be possible to require that all administrators use it.
 
-Similar to Multi-Factor Authentication this defence can be combined with other techniques to ask user to solve CAPTCHA only in special situations. 
+## Secondary Passwords, PINs and Security Questions
 
-Lots of CAPTCHA examples can be reviewed [here](https://www.whoishostingthis.com/resources/captcha/)
+As well as requiring a user to enter their password when authentication, they can also be prompted to provide additional security information such as:
 
-## Defense Option 3: IP blacklists
+- A PIN
+- Specific characters from a secondary passwords or memorable word
+- Answers to security questions
```

Comment:
Can be interesting to add a link to the [Choosing and Using Security Questions Cheat Sheet](Choosing_and_Using_Security_Questions_Cheat_Sheet.md).

Reply:
Added in 03ebaabe3b903b0917ccef93f02b451329fb228a.